tests/foreman/ui/test_rhai.py

```diff
@@ -0,0 +1,166 @@
+                activation_key.add_subscriptions({
+                    'quantity': 1,
+                    'subscription_id': subscription['id'],
+                })
+
+            # Attach custom product subscription to the activation key
+            if subscription['product_name'] == custom_product.name:
+                activation_key.add_subscriptions({
+                    'quantity': 1,
+                    'subscription_id': subscription['id'],
+                })
+                break
+
+        # Create VM
+        with VirtualMachine(distro='rhel66') as vm:
+            # Download and Install ketello-ca rpm
+            result = vm.run(
+                "wget -nd -r -l1 --no-parent -A '*.noarch.rpm' http://{0}/pub/"
+                .format(self.server_name))
+            self.assertEqual(result.return_code, 0)
+            result = vm.run('rpm -i katello-ca-consumer*.noarch.rpm')
+            self.assertEqual(
+                result.return_code, 0,
+                "failed to install katello-ca rpm: {0}, return code: {1}"
+                .format(result.stderr, result.return_code)
+            )
+            # Register client with foreman server using activation-key
+            result = vm.run(
+                'subscription-manager register --activationkey {0} '
+                '--org {1} --force'
+                .format(activation_key_name, org.label)
+            )
+
+            # Red Hat Access Insights requires RHEL 6/7 repo and it not
+            # possible to sync the repo during the tests,
+            # adding a file in /etc/yum.repos.d/rhel6/7.repo
+
+            rhel6_repo = conf.properties.get('insights.rhel6_repo')
+
+            repo_file = (
+                '[rhel6-rpms]\n'
+                'name=RHEL6\n'
+                'baseurl={0}\n'
+                'enabled=1\n'
+                .format(rhel6_repo)
+            )
+
+            vm.run(
+                'echo "{0}" >> /etc/yum.repos.d/rhel6.repo'
+                .format(repo_file)
+            )
+
+            # Install redhat-access-insights package
+            package_name = 'redhat-access-insights'
+            result = vm.run('yum install -y {0}'.format(package_name))
+            self.assertEqual(result.return_code, 0)
+
+            # Verify if package is installed by query it
+            result = vm.run('rpm -q {0}'.format(package_name))
+            self.assertEqual(result.return_code, 0)
+
+            # Register client with Red Hat Access Insights
+            result = vm.run('redhat-access-insights --register')
+            self.assertEqual(result.return_code, 0)
+
+        with Session(self.browser) as session:
+            # view clients registered to Red Hat Access Insights
+            session.nav.go_to_select_org(org.name)
+            Navigator(self.browser).go_to_insights_systems()
+            result = self.rhai.view_registered_systems()
+            self.assertIn("1", result, "Registered clients are not listed")
```

Comment:
The message `, "Registered clients are not listed"` is not required the code itself and the logs will tell this.